agency-api-spec.md

```diff
@@ -157,6 +157,118 @@ JSON payload that contains the form fields.
+**Required:** | if applicable
+**Example:** | `{"form_460A_case_number": "3347", "requester_type": "Journalist"}`
+
+
+#### Agency form fields
+
+Your agency might have additional fields specified in your [agency metadata file](https://github.com/18F/foia-recommendations/blob/master/schemas.md#agency-metadata-file).
+These additional fields are unique to your agency and are captured separately in
+the `agency_form_fields` object.
+
+The specific fields in `agency_form_fields` will be defined by your metadata
+file which includes both required and optional form fields. Any fields in
+`required_form_fields` will be considered required. Any fields in
+`additional_form_fields` will be considered optional. The portal will validate
+that any required fields are present.
+
+**Field:** | `*`
+:--- |:---
+**Type:** | determined by the [agency metadata file][agency-metadata-file-schema]
+**Description:** | Agency specific request form field as specified in your [agency's metadata file][agency-metadata-file-schema].
+**Required:** | if applicable
+**Example:** | See [below](#agency-form-fields-example)
+
+
+<a id="agency-form-fields-example"></a>
+##### Example
+
+Consider this sample [agency metadata
+file](https://github.com/18F/foia/blob/master/GSA.json). A truncated version is
+provided below.
+
+```
+{
+    "abbreviation": "GSA",
+    "departments": [
+        {
+            // ...
+            "additional_form_fields": [
+                {
+                    "help_text": "If your request relates to a GSA contract, please provide the contract number (which starts with \"GS-\")",
+                    "label": "GS- Contract number",
+                    "name": "contract_number"
+
+                },
+                {
+                    "help_text": "(i.e. New England Region (1A) - States Served: CT, MA, ME, NH, RI, VT",
+                    "label": "GSA Region",
+                    "name": "region"
+                }
+            ],
+            "required_form_fields": [
+                {
+                    "enum": [
+                        "Company",
+                        "Individual/Self",
+                        "Organization"
+                    ],
+                    "help_text": "Company",
+                    "label": "Request Origin",
+                    "name": "request_origin",
+                    "regs_url": null
+                }
+            ]
+        }
+    ]
+}
+```
+
+`agency_form_fields` therefore has the following fields.
+
+- (required) `request_origin`
+- `contract_number`
+- `region`
+
+So in the request payload, `agency_form_fields` would look like:
+
+```
+{
+    // ...
+    "agency_form_fields": {
+        "contract_number": "5547",
+        "region": "9",
+        "request_origin": "Individual/Self"
+    }
+}
+```
+
+**Field:** | `request_origin`
```

Comment:
My initial reactive to having these agency-specific fields listed here in addition to the examples above is that this more formal presentation of them might give the impression that they're required universally. 

I think your metadata to JSON example above sufficiently conveys how these agency-specific questions will be handled, so I'd be inclined to remove the tables below (fair warning: when on the fence, I always default to less instead of more).